scripts/image/dither.m

```diff
@@ -0,0 +1,411 @@
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with this program; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn  {Function File} {@var{X} = } dither (@var{RGB}, @var{map})
+## @deftypefnx {Function File} {@var{X} = } dither (@var{RGB}, @var{map}, @var{Qm}, @var{Qe})
+## @deftypefnx {Function File} {@var{BW} = } dither (@var{I})
+## Quantize an image using dithering to increase the apparent color resolution.
+##
+## @code{@var{X} = dither (@var{RGB},@var{map})} creates an indexed image
+## approximation.  It uses the color provided in the colormap, and uses
+## dithering to increase apparent color resolution.  Floyd-Steinberg error
+## filter is:
+## @code{[   x  7]}
+## @code{[3  5  1] / 16}
+## It uses a raster scan and no weight renormalization at boundaries.
+## The default values are used: @var{Qm}=5 and @var{Qe}=8.
+##
+## @var{RGB} is a mxnx3 array with values in @code{[0, 1]} (double) or
+## @code{[0, 255]} (uint8).
+##
+## @var{map} is cx3 matrix holding RGB triplets in @code{[0, 1]} (double).
+##
+## @var{Qm} is the number of quantization bits per axis for inverse colormap
+## (default: 5).
+##
+## @var{Qe} is the number of quantization bits for error diffusion (default: 8,
+## max 16).
+##
+## @var{X} is a mxn indexed image (uint8 if c<=256, else uint16) for the
+## colormap @var{map} provided.
+##
+## Example:
+## @example
+## X = dither (RGB, map);
+## @end example
+##
+## @code{@var{X} = dither (@var{RGB}, @var{map}, @var{Qm}, @var{Qe})}
+##
+## @var{Qm} is the number of quantization bits along each color axis for the
+## inverse colormap.  @var{Qm} determines the resolution of this grid along each
+## color axis (R, G, B).  @var{Qm} defines the precision of the color space
+## discretization used to map input RGB values to those colors available in the
+## colormap.  @var{Qe} is the number of quantization bits for the color space
+## error calculations in the Floyd-Steinberg error diffusion algorithm.
+## It controls the precision of the error values that are calculated and
+## propagated during dithering.  If @var{Qe} < @var{Qm}, the error diffusion
+## process may lose precision.  Therefore dithering cannot be performed, and the
+## function returns an undithered indexed image.
+##
+## @code{@var{BW} = dither (@var{I})} converts the grayscale input image @var{I}
+## into binary applying dithering in the process.  The output image @var{BW}
+## is a black and white image where dithering creates the illusion of shades of
+## gray.
+##
+## @cite{Floyd, R. W., and Steinberg, L., An Adaptive Algorithm for Spatial Gray
+## Scale, International Symposium Digest of Technical Papers, Society for
+## Information Displays, 1975, p. 36.}
+## @cite{Ulichney, R., Digital Halftoning, The MIT Press, 1987.}
```

Comment:
This renders like the following as PDF:

<img width="1147" height="115" alt="Image" src="https://github.com/user-attachments/assets/27b7c13e-e70e-4b97-85be-98ea97799666" />

Maybe something along the lines what is done in condest.m would look better?
https://github.com/gnu-octave/octave/blob/81aa7937c6a42a079702654b4f0daef75928665f/scripts/linear-algebra/condest.m#L98-L111